src/microsAuthentication/content.md

```diff
@@ -0,0 +1,1157 @@
+
+```terminal|command=1|title=bash
+kubectl --namepsace test exec -ti curl bash
+root@curl:/#
+```
+
+Inside the shell session, first export a few variables.
+
+First up, let's export the APISERVER variable pointing to the Kubernetes API:
+
+```terminal|command=1|title=bash
+# APISERVER=https://kubernetes.default.svc
+```
+
+The SERVICEACCOUNT variable points to the filesystem directory at which the service account related files can be found:
+
+```terminal|command=1|title=bash
+# SERVICEACCOUNT=/var/run/secrets/kubernetes.io/serviceaccount
+```
+
+The NAMESPACE variable contains the namespace obtained via reading the contents of the namespace file in the above directory:
+
+```terminal|command=1|title=bash
+# NAMESPACE=$(cat ${SERVICEACCOUNT}/namespace)
+```
+
+The TOKEN variable contains the service account token obtained via reading the contents of the token file in the same directory:
+
+```terminal|command=1|title=bash
+# TOKEN=$(cat ${SERVICEACCOUNT}/token)
+```
+
+The CACERT variable points to the Certificate Authority bundle that curl will use to verify the Kubernetes API certificate:
+
+```terminal|command=1|title=bash
+# CACERT=${SERVICEACCOUNT}/ca.crt
+```
+
+Let's issue a request to the Kubernetes API and see if you can authenticate:
+
+```terminal|command=1|title=bash
+curl --cacert ${CACERT} --header "Authorization: Bearer ${TOKEN}" -X GET ${APISERVER}/api
+{
+  "kind": "APIVersions",
+  "versions": [
+  "v1"
+  ],
+  "serverAddressByClientCIDRs": [
+  {
+    "clientCIDR": "0.0.0.0/0",
+    "serverAddress": "172.17.0.3:8443"
+  }
+  ]
+}
+```
+
+Excellent, you can reach the Kubernetes API.
+
+Note how you passed the Authorization header containing the service account token.
+
+If you don't supply it or pass in an invalid token, you will get a HTTP 403 error response as follows:
+
+```terminal|command=1|title=bash
+curl --cacert ${CACERT} -X GET ${APISERVER}/api
+{
+  "kind": "Status",
+  "apiVersion": "v1",
+  "metadata": {},
+  "status": "Failure",
+  "message": "forbidden: User \"system:anonymous\" cannot get path \"/api\"",
+  "reason": "Forbidden",
+  "details": {},
+  "code": 403
+}
+```
+
+Next, you will see how you can invoke the `TokenReview API` to verify a service account token.
+
+### Sending a token verification request
+
+To issue a request to the TokenReviewAPI, you need two parts:
+
+1. The API endpoint that you should call which is `/apis/authentication.k8s.io/v1/tokenreviews`.
+1. A payload with the information about the token that you wish to validate.
+
+The API endpoint expects a POST request with a JSON formatted payload of the following form:
+
+```json
+{
+  "kind": "TokenReview",
+  "apiVersion": "authentication.k8s.io/v1",
+  "spec": {
+  "token": "<your token>"
+  }
+}
+```
+
+In this case, the token you will use to authenticate to the API server as well as verify the status
+of is the same. Thus, the curl command to run in the pod shell session is:
```

Comment:
Is this sentence necessary? When is the token not the same?
I'm a bit confused since the assumption is that I need only 1 token.

Reply:
@danielepolencic Consider the the 2nd demo - where the secret store is going to verify whether the token presented to it via the calling service (API) is legit or not. The secret store uses its own SA token to authenticate itself to the API when it makes a TokenReview request to validate the API token. In that case, the token being verified is different from the token being used to authenticate itself.

Reply:
I'm tempted to explain that when we get to it.
It's easier if we make the content progressively more complex than presenting all the complexity upfront.
Thoughts?